azure-pipelines.yml

```diff
@@ -174,22 +180,27 @@ stages:
             testQPY: false
             testImages: true
             installFromSdist: true
+            installOptionals: false
```

Comment:
One mild concern I have here is when we're adding support for a new python version it might be harder to know when our optional extras support it with this since we're only running the old side with the optional extras. That being said I think we should be ok to try this to start and we'll see how it goes for Python 3.12 support.

Reply:
I set the cronjob CI to use the full optionals, so we'll find out from the nightly run whether stuff works or not without blocking CI.